### DISCUSSION TITLE
Energy Meter Tuya - integrarlo in HA

### DISCUSSION THREAD

**Discussion by Matty1184** [indomus/forum#2620] [GENERALE - Discussioni generiche]
Ciao a tutti, 
ho preso un dispositivo energy meter wi-fi (sbagliando pure l'ordine perchè lo volevo zigbee!), integrato provvisoriamente in tuya smart per verificare il funzionamento etc etc, ora stavo guardando di integrarlo in HA e il mio scopo era non farlo girare tramite cloud ma in locale.
se non ho letto male si dovrebbe usare Tuya local tramite HACS ma se non ho capito male l'account developer di tuya dura solo 1 mese.
scaduto questo mese, se il dispositivo è collegato se non ho capito male ok non succede nulla, ma se devo aggiungere un dispositivo, devo creare un nuovo account?
soluzioni consigliate diverse?

Ringrazio tutti i n anticpo.

**Comment by 50494554524F** [comment]
nell'account developer (cmq dovrebbe rinnovarsi gratuitamente) puoi prendere tutti i dati dei dispositivi da usare poi con tuya local... su come fare devi arrangiarti perché non uso tuya da un po' e non mi ricordo

**Comment by johnnyrun** [comment]
Puoi usare lo stesso account per aggiungere altri dispositivi. in sostanza aggiungi nell'app e poi usi in locale.
Un pò difficoltoso reperire la chiave ma poi va..

Consiglio: se puoi associa riserva l'ip locale... ti togli qualche mal di testa

Ciao

**Comment by Matty1184** [comment]
> 
> Consiglio: se puoi associa riserva l'ip locale... ti togli qualche mal di testa

Intendi associare un ip statico tramite macid nel router?

una cosa strana  che ho notato è che il dispositivo interrogato da applicazione tuya ha un ip su una subnet diversa dalla mia rete, non so se questo è dovuto dalla gestione dell'app (lo spero)

**Comment by johnnyrun** [comment]
sì, intendo DHCP reservation.

L'ip che vedi da tuya smart, dovrebbe essere il tuo ip pubblico

**Comment by Matty1184** [comment]
ciao a tutti, ho finalmente trovato il tempo per integrare il dispositivo ma sto riscontrando alcuni problemi:
1) il dispositivo spesso è offline sia da app tuya, che da pannello IoT tuya. non riesco a capire perchè nonostante vi sia un consumo, lui resti offline. quando riesco a farlo andare online (non faccio nulla ed è casuale il suo stato) sul pannello device di tuya è sempre offline.
2) nelle guide che ho letto consigliano di sottoscrivere l'api "Smart Home Scene Linkage" ma sul sito tuya risulta "deprecated".se non ho capito male la nuova è incluca nella "IoT Core". corretto?
